### ISSUE TITLE
crashes when a font is not found

### ISSUE THREAD

**Issue by kilobyte** [tomszilagyi/zutty#59]
When the requested font is not found, zutty crashes:
```
[/tmp/zutty](master)$ build/src/zutty -font blargh
E [fontpack.cc:218] Error: No Regular variant of the requested font 'blargh' could be identified.
terminate called after throwing an instance of 'std::runtime_error'
  what():  No suitable files for 'blargh' found!
Aborted (core dumped)
```
Same happens for Debian's version 0.10.20211016.133334+dfsg1-1 and for self-built git master.

**Comment by tomszilagyi** [comment]
What would you like Zutty to do, exactly?

**Comment by kilobyte** [comment]
Give an error message and cleanly exit.

**Comment by tomszilagyi** [comment]
I think the error message is pretty darn clear: `Error: No Regular variant of the requested font 'blargh' could be identified.`
What is the importance of a clean exit to you?

**Comment by kilobyte** [comment]
The reasons are twofold:
1. making the user believe zutty crashed
* that the search went wrong (instead of simply not finding the font)
* that it's a bug in the program (as no crashes is the usual policy)
2. triggering the system's crash handler if enabled
* (for systemd users) systemd-coredump will log the crash in system log and save the core
* some distributions (Ubuntu?) are configured to spawn a crash reporter that asks the user whether to submit a bug
* old-style Unix folks like me will get the core dump in $CWD

**Comment by tomszilagyi** [comment]
I see no business value in ensuring a clean exit from a state that is clearly FUBAR. The user should be intelligent enough to understand an error message in plain English, even if it is wrapped in something that looks like a "crash".

Of course, technically Zutty does not crash per se, rather it exits with an abnormal condition (uncaught exception) from a runtime detected state that is impossible to handle in any reasonable way (within the program). Whether this is something triggered by an overly clever user (like in this case) or a genuine programming error (bug) that needs attention is not easily discriminated *within the program*.

In general, you either (knowingly) did what you did, in which case my response is "so don't do it". Or there is a bug (yes, you are correct, Zutty should not crash unless provoked, but I have no proofs that it won't, so it might), in which case I am very interested in all the evidence. My point is that while it is probably trivial to tell these cases apart when you look at it as a human, it is very far from trivial to tell within the program.

A clean exit would possibly mask things and then people would complain about Zutty "not doing anything" (starting and disappearing, without a trace if they did not start it from another terminal to observe its output) instead of crashing.

As a final note, correct "handling" of these exceptions would add complexity, because Zutty is a multithreaded program, so we would need to ensure that threads do not get stuck, etc. As is, this cleanup is very robust, done by the OS for free.

**Comment by kilobyte** [comment]
An uncaught exception is a bug.  But hey, I don't care that much — I'm arguing for the sake of newbie distro users who will get that crash handler that asks whether to submit a dump.  Just commenting for the next case someone reports. 😉

In a multithreaded program, you can get the same effect by calling `_exit()` (aka `exit_group()`) which skips all the libc handlers and goes straight to the OS.
